smarts/core/agent_interface.py

```diff
@@ -129,6 +129,23 @@ class AgentType(IntEnum):
     """Agent performs trajectory tracking using model predictive control."""
 
 
+@dataclass(frozen=True)
+class AgentsAliveDoneCriteria:
+    total_ego_agents_alive: Optional[int] = None
+    """if set, trigggers the agent to be done if total number of alive ego agents falls below total_agents_alive"""
+    total_social_agents_alive: Optional[int] = None
```

Comment:
I wonder if we should make distinction between social agents and agents here?

Reply:
so instead of having `minimum_social_agents_alive` have `minimum_total_agents_alive`?
I think both approaches are pretty similar, if the user wants X number of ego agents alive while keeping Y number of social agents alive, the user could do `minimum_ego_agents_alive=X` `minimum_total_agents_alive=X+Y` or `minimum_ego_agents_alive=X` `minimum_social_agents_alive=Y`. 
I guess there is a tradeoff between if the user wants just X number of total agents alive or just Y number of social agents alive. Perhaps it is better to just add another option for "minimum_total_agents_alive"?

Reply:
Sure, my consideration here was just that the number of ego agents is important. I am no not so much sure that the number of social agents is important by itself because in general we treat social agents as just agents but perhaps at a finalised stage. I also note the hole here in that there is no current way to end on reaching a minimum number of total agents.

I think the options I would prefer here is to remove the option so we do not have to explicitly support the social agent distinction or to go with the `minimum_total_agents_alive=X+Y` option.

Reply:
got it, added the `minimum_total_agents_alive` option